packages/plugins/typed-variable-modal/src/TypedVariableModal.js

```diff
@@ -0,0 +1,385 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class responsible for creating a modal used for creating typed
+ *     variables.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+
+import * as Blockly from 'blockly/core';
+import { Modal } from './Modal.js';
+
+/**
+ * Class for displaying a modal used for creating typed variables.
+ */
+export class TypedVariableModal extends Modal {
+  /**
+   * Constructor for creating a typed variable modal.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the modal will
+   *     be registered on.
+   * @param {string} btnCallbackName The name used to register the button
+   *     callback.
+   * @param {Array<Array<string>>} types An array holding arrays with the
+   *     display name as the first value and the type as the second.
+   *     Ex: [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']].
+   * @param {TypedVarModalMessages} opt_messages The messages for a typed variable
+   *     modal.
+   */
+  constructor(workspace, btnCallbackName, types, opt_messages) {
+    const title = opt_messages ? opt_messages["TYPED_VAR_MODAL_TITLE"] :
```

Comment:
If you're passed in a opt_messages bag that doesn't contain 'TYPED_VAR_MODAL_TITLE', you get no title. 

Alternately you could do: 
```
(opt_messages && opt_messages['TYPED_VAR_MODAL_TITLE']) || 'Create Typed Variable';
```